### PR TITLE
[MXNET-1421] Added (CuDNN)BatchNorm operator to the list of mirrored operators

### PR DESCRIPTION
**Pull request by ArmageddonKnight**
## Description ##
Added the (CuDNN)BatchNorm operator to the list of mirrored operators.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] In file `src/executor/graph_executor.cc` method `GraphExecutor::InitFullGraph`, removed the restriction on the `(CuDNN)BatchNorm` operator.
- [x] In file `src/operator/cudnn_batch_norm-inl.h` class `CuDNNBatchNormOp`, added an internal lock on auxiliary states to prevent multiple consecutive updates on the moving mean and variance.

## Comments ##

I have documented the changes [**here**](https://v2.overleaf.com/read/snhdnvhmjrcb). Any comment is welcome. Thanks.

FYI, @antinucleon 

S.A. Issue #14383

### CODE REVIEW

**Comment by eric-haibin-lin** [comment]
@antinucleon

**Comment by eric-haibin-lin** [comment]
@ArmageddonKnight can you look into the sync batch norm failure?

**Comment by ArmageddonKnight** [comment]
@eric-haibin-lin I think I fixed the `sync_batch_norm` test failure.

**Comment by eric-haibin-lin** [comment]
There was some update in mxnet website build. Could you sync with mxnet master? thanks
